deployment/configenv/xml_jlibpt/EnvHelper.hpp

```diff
@@ -0,0 +1,121 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+#ifndef _ENVHELPER_HPP_
+#define _ENVHELPER_HPP_
+
+#include "jiface.hpp"
+#include "jliball.hpp"
+#include "XMLTags.h"
+//#include "IConfigEnv.hpp"
+#include "ConfigEnvError.hpp"
+#include "GenEnvRules.hpp"
+
+namespace ech
+{
+
+//#define ATTR_SEP  "^"
+//#define ATTR_V_SEP "|"
+
+//interface IConfigEnv;
+
+//interface IConfigComp;
+interface IConfigComp : public IInterface
+{
+  virtual void create(IPropertyTree *params) = 0;
+  virtual unsigned add(IPropertyTree *params) = 0;
+  //virtual void addNode(IPropertyTree *node, const char* xpath, bool merge) = 0;
+  virtual void modify(IPropertyTree *params) = 0;
+  virtual void remove(IPropertyTree *params) = 0;
+};
+
+#define DEFAULT_ENV_XML CONFIG_DIR"/environment.xml"
+#define DEFAULT_ENV_OPTIONS CONFIG_DIR"/environment.conf"
+#define DEFAULT_GEN_ENV_RULES CONFIG_DIR"/genenvrules.conf"
+#define DEFAULT_BUILDSET COMPONENTFILES_DIR"/configxml/buildset.xml"
+#define ESP_CONFIG_PATH INSTALL_DIR "" CONFIG_DIR "/configmgr/esp.xml"
+
+#define USE_WIZARD 1
+
+enum CONFIG_TYPE { CONFIG_INPUT, CONFIG_ENV, CONFIG_ALL };
```

Comment:
Why not put the CONFIG_TYPE enum inside the class as a public. The refer to it as EnvConfigOptions::<enum>. That way there is no potential conflict with enum values elsewhere.